frappe/public/js/frappe/form/controls/table_multiselect.js

```diff
@@ -81,28 +81,36 @@ frappe.ui.form.ControlTableMultiSelect = class ControlTableMultiSelect extends (
 		}
 
 		const link_field = this.get_link_field();
+		// Trim the value to remove spaces or only if space is only input
+		if (value && typeof value === "string") {
+			value = value.trim();
+		}
 
 		if (value) {
-			if (this.frm) {
-				const new_row = frappe.model.add_child(
-					this.frm.doc,
-					this.df.options,
-					this.df.fieldname
-				);
-				new_row[link_field.fieldname] = value;
-				this.rows = this.frm.doc[this.df.fieldname];
-
-				this.frm.script_manager.trigger(
-					`${this.df.fieldname}_add`,
-					this.df.options,
-					new_row.name
-				);
-			} else {
-				this.rows.push({
-					[link_field.fieldname]: value,
-				});
+			// Only create a pill if the value is a real item from the autocomplete list.
```

Comment:
This part seems unrelated to the PR description - can you shift it into a separate PR, seems like a UX improvement.